### ISSUE TITLE
[CM-10] Charger: flashing white screen

### ISSUE THREAD

**Issue by ItachiSan** [TheWhisp/android_device_samsung_msm7x27a-common#210]
As in title.

**How to reproduce**

1) Turn off phone
2) Plug it to charger
3) Instead of a single blink or none (single in CM-10.1, none in CM-11,0), screen will be turning on and off in a completily white image.

**Expected**
To blink once or not at all.

**Thoughts**
Is maybe a _init_ binary problem, that has been fixed later...?

**Comment by TheWhisp** [comment]
no, it has nothing to do with the init binary. Either kernel or miniui (graphics.c)

**Comment by ItachiSan** [comment]
Kernel is the same of CM-10.1...
So, maybe is _graphics.c_

**Comment by TheWhisp** [comment]
init only detects the boot mode, if it failed to do that, it wouldn't even be possible to boot into charging mode.

**Comment by ItachiSan** [comment]
Actually, it detects it (so it seems, it pauses boot in this infinite blinking that should be the battery charge animation)

**Comment by ItachiSan** [comment]
Trying with this _graphics.c_:
https://github.com/ItachiSan/android_device_samsung_msm7x27a-common/blob/jellybean-ion/recovery/graphics.c
Will report to you later.

**Comment by TheWhisp** [comment]
I know it detects charging mode, however, I am too lazy to compile and flash cm-10 :P

**Comment by ItachiSan** [comment]
LOL
Already doin' it sir! ;)

**Comment by ItachiSan** [comment]
Update: now the screen does not blink...
It become black and do nothing at all. LOL
Mmmmmh... Maybe I could use CM-10.1 recovery as repo to build against CM-10.

**Comment by ItachiSan** [comment]
Nothing to do.
Actually, the thing I know is that the function that makes the screen blinking is
_void gr_fb_blank(bool blank)_
but it's kinda useless.

Now I'm confused.
- minui is almost the same between cm-10 and cm-10.1
- the kernel is the same for cm-10 and cm-10.1 (used _cm-10.1-new_ branch)

So, what's wrong?
Could be anything else than kernel or minui?

**Comment by ItachiSan** [comment]
CRAP
I think I found the solution.

**Comment by TheWhisp** [comment]
yes, it could also be the charger binary itself

**Comment by ItachiSan** [comment]
And it is.
In fact,it should be patches (for CM-10.1 I mean patch 3 and 4 for _system_core_)

**Comment by ItachiSan** [comment]
Actually fixed with this commit:
https://github.com/ItachiSan/android_device_samsung_msm7x27a-common/commit/e39643c6fd836a63f313c2e273d5b94511779bda
Will make a branch with just the patches and then open a pull request.
As for now, closing the issue as it's solved.

**Comment by ItachiSan** [comment]
I had a working _boot.img_ with kernel branch _cm-10.1_ (non-ion one), with _cm-10.1-new_ I've the same problem as before. Doh

**Comment by ItachiSan** [comment]
Seems that I've found a solution, need more tests

**Comment by ItachiSan** [comment]
Solution was wrong. ><
Can you help me? Could it be kernel related?
Considering the fact that with old branch it was working and not with new one...

**Comment by ItachiSan** [comment]
Tried debugging charger.
Actually, this is the **kmsg**
https://gist.github.com/ItachiSan/8782049
Share your ideas here below (after some tests for Wifi tethering, I think I'll switch to CM-10.1, have to relax lol)

**Comment by TheWhisp** [comment]
cm-10.1 branch is based on kernel with older display drivers, so if it was working on you, you should check /drivers/video/msm

**Comment by ItachiSan** [comment]
Thanks mate! Will check around :)

**Comment by TheWhisp** [comment]
more specifically, if I am not mistaken, in msm_fb.c

**Comment by ItachiSan** [comment]
Trying to make a **_boot.img**_ with _cm-10.1_ branch _msm_fb.c_

**Comment by ItachiSan** [comment]
Done, still white screen.
Switching to CM-10.1, so I can fix the bugs I've not found or fixed in CM-10

**Comment by bacevs** [comment]
any progess ItachiSan?

**Comment by ItachiSan** [comment]
I found that _cm-10.1_ kernel branch gives no flashing screen...
So I think I'll take video drivers for it (ASAP obviously)
